src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -576,6 +619,124 @@ public void removeElementFilesAndLangKeys(ModElement element) {
 		return new ArrayList<>(files);
 	}
 
+	public List<GeneratorTemplatesList> getModElementListTemplates(ModElement element) {
+		return getModElementListTemplates(element, false, null);
+	}
+
+	public List<GeneratorTemplatesList> getModElementListTemplates(ModElement element,
+			GeneratableElement generatableElement) {
+		return getModElementListTemplates(element, false, generatableElement);
+	}
+
+	private List<GeneratorTemplatesList> getModElementListTemplates(ModElement element, boolean performFSTasks,
+			GeneratableElement generatableElement) {
+
+		if (generatableElement == null) {
+			generatableElement = element.getGeneratableElement();
+			if (generatableElement == null) { // we can't construct list data because we have nothing to process
+				LOG.warn("Failed to load mod generatable element: " + element.getName()
+						+ ". This means no list templates will be generated");
+				return new ArrayList<>();
+			}
+		}
+
+		Map<?, ?> map = generatorConfiguration.getDefinitionsProvider().getModElementDefinition(element.getType());
+		if (map == null) {
+			LOG.info("Failed to load element definition for mod element type " + element.getType().getRegistryName());
+			return null;
+		}
+
+		Set<GeneratorTemplatesList> fileLists = new HashSet<>();
+		final GeneratableElement generatable = generatableElement;
+
+		List<?> templateLists = (List<?>) map.get("list_templates");
+		if (templateLists != null) {
+			int templateID = 0;
+			for (Object list : templateLists) {
+				Map<GeneratorTemplate, List<Boolean>> files = new LinkedHashMap<>();
+				String groupName = (String) Objects.requireNonNullElse(((Map<?, ?>) list).get("name"),
+						"Templates group");
+				Object listData = TemplateExpressionParser.processFTLExpression(this,
+						(String) ((Map<?, ?>) list).get("listData"), generatableElement);
+				List<?> templates = (List<?>) ((Map<?, ?>) list).get("forEach");
+				List<?> elements = new ArrayList<>();
+				if (listData instanceof Map<?, ?> listMap)
+					elements = new ArrayList<>(listMap.entrySet());
+				else if (listData instanceof Collection<?> collection)
+					elements = new ArrayList<>(collection);
+				else if (listData instanceof Iterable<?> iterable) // fallback for the worst case
+					elements = new ArrayList<>(StreamSupport.stream(iterable.spliterator(), false).toList());

```

Comment:
(TODO) In case we switch to J18 (J17-preview) before this PR is merged:
```suggestion
				List<?> elements = switch (listData) {
					case Map<?, ?> listMap -> new ArrayList<>(listMap.entrySet());
					case Collection<?> collection -> new ArrayList<>(collection);
					// fallback for the worst case
					case Iterable<?> iterable -> new ArrayList<>(StreamSupport.stream(iterable.spliterator(), false)
						.toList());
					case null, default -> new ArrayList<>();
				};
```